api/tests.py

```diff
@@ -1,3 +1,158 @@
+        self.project = ProjectFactory()
+        self.project.geo = self.geometry_store
+        self.project.save()
+        self.user = User(username='test', password='test')
+        self.user.save()
+
+    def test_published_project(self):
+        self.project.published = True
+        self.project.save()
+
+        with self.settings(PUBLISH_FILTER_ENABLED=True):
+            with self.subTest('Authenticated and PUBLISH_FILTER_ENABLED'):
+                self.client.force_login(self.user)
+                response = self.client.get(self.url)
+                self.assertEqual(response.status_code, 200)
+
+            with self.subTest('Not authenticated, PUBLISH_FILTER_ENABLED'):
+                self.client.logout()
+                response = self.client.get(self.url)
+                self.assertEqual(response.status_code, 200)
+
+        with self.settings(PUBLISH_FILTER_ENABLED=False):
+            with self.subTest('Authenticated and PUBLISH_FILTER_ENABLED == False'):
+                self.client.force_login(self.user)
+                response = self.client.get(self.url)
+                self.assertEqual(response.status_code, 200)
+
+            with self.subTest('Not authenticated, PUBLISH_FILTER_ENABLED == False'):
+                self.client.logout()
+                response = self.client.get(self.url)
+                self.assertEqual(response.status_code, 200)
+
+    def test_unpublished_project(self):
+        self.project.published = False
+        self.project.save()
+
+        with self.settings(PUBLISH_FILTER_ENABLED=True):
+            with self.subTest('Authenticated and PUBLISH_FILTER_ENABLED'):
+                self.client.force_login(self.user)
+                response = self.client.get(self.url)
+                self.assertEqual(response.status_code, 200)
+
+            with self.subTest('Not authenticated, PUBLISH_FILTER_ENABLED'):
+                self.client.logout()
+                response = self.client.get(self.url)
+                self.assertEqual(response.status_code, 404)
+
+        with self.settings(PUBLISH_FILTER_ENABLED=False):
+            with self.subTest('Authenticated and PUBLISH_FILTER_ENABLED == False'):
+                self.client.force_login(self.user)
+                response = self.client.get(self.url)
+                self.assertEqual(response.status_code, 200)
+
+            with self.subTest('Not authenticated, PUBLISH_FILTER_ENABLED == False'):
+                self.client.logout()
+                response = self.client.get(self.url)
+                self.assertEqual(response.status_code, 200)
+
+
+class TestGeometryStoreCentroidViewSet(TestCase):
+
+    def setUp(self):
+        self.geom_with_published_project = GeometryStore()
+        self.geom_with_published_project.save()
+        point = PointGeometry(geom=Point(20, 30))
+        point.save()
+        self.geom_with_published_project.points.add(point)
```

Comment:
Similar feedback here. Why is all of these setup needed? It doesn't appear to be used in the tests.